core/server/master/src/test/java/alluxio/master/block/BlockMasterMetricsTest.java

```diff
@@ -73,6 +75,14 @@ public void testMetricsTierCapacity() {
         getGauge(MetricKey.CLUSTER_CAPACITY_FREE.getName() + MetricInfo.TIER + HDD));
   }
 
+  @Test
+  public void testSize() {
+    BlockStore blockStore = Mockito.mock(BlockStore.class);
+    when(blockStore.size()).thenReturn(100L);
+    Whitebox.setInternalState(mBlockMaster, "mBlockStore", blockStore);
```

Comment:
whitebox is hard to maintain. How about introducing a new constructor so we can inject the dependency of `blockStore`?

```java
@VisibleForTesting
DefaultBlockMaster(MetricsMaster metricsMaster, CoreMasterContext masterContext, Clock clock,
    ExecutorServiceFactory executorServiceFactory, BlockStore blockStore) {
  mBlockStore = blockStore
  ...
}
```

Reply:
for this, since the BlockMaster was a mocked class, whitebox was the only way to do this. otherwise we will need to construct metricsMaster, masterContext etc.